examples/example.c

```diff
@@ -403,8 +418,15 @@ main(int argc, char **argv)
         if (has_arg(argc, argv, "unsample-tx")) {
             sentry_transaction_context_set_sampled(tx_ctx, 0);
         }
+        sentry_value_t custom_sampling_ctx = sentry_value_new_object();
+        sentry_value_set_by_key(
+            custom_sampling_ctx, "a", sentry_value_new_string("first_value"));
+        sentry_value_set_by_key(
+            custom_sampling_ctx, "b", sentry_value_new_int32(42));
+        sentry_value_incref(custom_sampling_ctx);
         sentry_transaction_t *tx
-            = sentry_transaction_start(tx_ctx, sentry_value_new_null());
+            = sentry_transaction_start(tx_ctx, custom_sampling_ctx);
+        sentry_value_decref(custom_sampling_ctx);
```

Comment:
ASAN complains about this object leaking memory, but I'm not sure if we can expect the users to always provide a `sentry_value_t` custom sampling context (currently that is the required argument type for the `transaction_start` because it was already that way before, but in the new sampling context struct we allow for `void *`.

I think we maybe should change the argument type for the custom context to also be `void *` then.

Reply:
Maybe I also misunderstood the design idea behind the custom context in the `traces_sampler` scenario. I thought of it more as a context-access parameter for the callback (similar to our `on_crash` and `before_sen` hook) rather than a value used in the context of the remaining performance environment. Maybe it should be a `sentry_value_t`.

Let's not fix the ASAN issue, but I'd like to review the custom context first. I might have given you the wrong design guidance here.

Reply:
For now we only allow for `sentry_value_t` type custom contexts. If there is every an ask for more flexibility, we can reconsider the `void *` implementation.